sdk/cosmos/azure-cosmos/src/main/java/com/azure/cosmos/implementation/http/ReactorNettyClient.java

```diff
@@ -98,29 +100,41 @@ private void configureChannelPipelineHandlers() {
         Objects.requireNonNull(request.httpMethod());
         Objects.requireNonNull(request.uri());
         Objects.requireNonNull(this.httpClientConfig);
-        if(request.getReactorNettyRequestRecord() == null) {
+        if(request.reactorNettyRequestRecord() == null) {
             ReactorNettyRequestRecord reactorNettyRequestRecord = new ReactorNettyRequestRecord();
             reactorNettyRequestRecord.setTimeCreated(Instant.now());
-            request.setReactorNettyRequestRecord(reactorNettyRequestRecord);
+            request.withReactorNettyRequestRecord(reactorNettyRequestRecord);
         }
 
         final AtomicReference<ReactorNettyHttpResponse> responseReference = new AtomicReference<>();
 
+        final Duration responseTimeout;
+        if (OperationType.QueryPlan.equals(request.operationType())) {
+            responseTimeout = Duration.ofSeconds(this.httpClientConfig.getConfigs().getQueryPlanResponseTimeoutInSeconds());
+        } else if (OperationType.AddressRefresh.equals(request.operationType())) {
+            responseTimeout = Duration.ofSeconds(this.httpClientConfig.getConfigs().getAddressRefreshResponseTimeoutInSeconds());
+        } else {
+            responseTimeout = Duration.ofSeconds(this.httpClientConfig.getConfigs().getDirectHttpsResponseTimeoutInSeconds());
+        }
+
+        logger.info("Request Operation Type is : {}", request.operationType());
+
         return this.httpClient
             .observe((connection, state) -> {
                 Instant time = Instant.now();
                 if(state.equals(HttpClientState.CONNECTED) || state.equals(HttpClientState.ACQUIRED)){
-                    request.getReactorNettyRequestRecord().setTimeConnected(time);
+                    request.reactorNettyRequestRecord().setTimeConnected(time);
                 } else if(state.equals(HttpClientState.CONFIGURED)){
-                    request.getReactorNettyRequestRecord().setTimeConfigured(time);
+                    request.reactorNettyRequestRecord().setTimeConfigured(time);
                 } else if(state.equals(HttpClientState.REQUEST_SENT)){
-                    request.getReactorNettyRequestRecord().setTimeSent(time);
+                    request.reactorNettyRequestRecord().setTimeSent(time);
                 } else if(state.equals(HttpClientState.RESPONSE_RECEIVED)){
-                    request.getReactorNettyRequestRecord().setTimeReceived(time);
+                    request.reactorNettyRequestRecord().setTimeReceived(time);
```

Comment:
same here, why change in api name ?

Reply:
@simplynaveen20  - this is to correct the naming of this API to be in consistent with rest of the APIs in `HttpRequest` class.